python/pyspark/sql/connect/client/reattach.py

```diff
@@ -18,12 +18,11 @@
 
 check_dependencies(__name__)
 
+from threading import RLock
 import warnings
 import uuid
 from collections.abc import Generator
 from typing import Optional, Dict, Any, Iterator, Iterable, Tuple, Callable, cast, Type, ClassVar
-from multiprocessing import RLock
-from multiprocessing.synchronize import RLock as RLockBase
 from multiprocessing.pool import ThreadPool
```

Comment:
Is it compatible to use the pool from multiprocessing but the lock from threading?

Reply:
Yeah, this is actually a thread pool but it's under `multiprocessing` for some reasons. I took a look into the codes inside before. It should be safe to use threading.Rlock.